src/Spd.Manager.Printing/PrintingManager.cs

```diff
@@ -41,29 +54,54 @@ public async Task<string> Handle(StartPrintJobCommand request, CancellationToken
-                    return await UpdateResultInEvent(printResponse, request.EventId, cancellationToken);
+                    ResultResponse result = _mapper.Map<ResultResponse>(printResponse);
+                    await UpdateResultInEvent(result, request.EventId, cancellationToken);
+                    return result;
                 };
             }
             catch (Exception ex)
             {
-                SendResponse result = new(null, JobStatus.Failed, $"{ex.Message} {ex.InnerException?.Message}");
-                return await UpdateResultInEvent(result, request.EventId, cancellationToken);
+                ResultResponse result = new()
+                {
+                    PrintJobId = null,
+                    Status = JobStatusCode.Fail,
+                    Error = $"{ex.Message} {ex.InnerException?.Message}"
+                };
+                await UpdateResultInEvent(result, request.EventId, cancellationToken);
+                return result;
             }
         }
 
-        return string.Empty;
+        return null;
     }
 
-    public async Task<PrintJobStatusResp> Handle(PrintJobStatusQuery request, CancellationToken cancellationToken)
+    public async Task<ResultResponse> Handle(PrintJobStatusQuery request, CancellationToken cancellationToken)
     {
-        var statusResponse = await _printer.Report(new BCMailPlusPrintStatusRequest(request.PrintJobId), cancellationToken);
-        var status = statusResponse.Status switch
+        //get event 
+        EventResp? eventResp = await _eventRepo.GetAsync(request.EventId, cancellationToken);
+        if (eventResp == null)
+            throw new ApiException(System.Net.HttpStatusCode.BadRequest, "Invalid eventqueue id.");
+        if (eventResp.JobId == null)
+            throw new ApiException(System.Net.HttpStatusCode.BadRequest, "Invalid eventqueue id.The event has never been executed. There is no job id.");
+
+        try
         {
-            JobStatus.Completed => PrintJobStatus.Completed,
-            JobStatus.Failed => PrintJobStatus.Failed,
-            _ => PrintJobStatus.InProgress,
-        };
-        return new PrintJobStatusResp(status, statusResponse.Error);
+            ReportResponse statusResponse = await _printer.Report(new BCMailPlusPrintStatusRequest(eventResp.JobId), cancellationToken);
+            ResultResponse result = _mapper.Map<ResultResponse>(statusResponse);
+            await UpdateResultInEvent(result, request.EventId, cancellationToken);
+            return result;
+        }
+        catch (Exception ex)
+        {
+            ResultResponse result = new()
+            {
+                PrintJobId = eventResp.JobId,
+                Status = JobStatusCode.Fail,
+                Error = $"{ex.Message} {ex.InnerException?.Message}"
+            };
+            //await UpdateResultInEvent(result, request.EventId, cancellationToken); //we should not update job status if we read status failed.

```

Comment:
Do we need this commented out code?

Reply:
leave it there. 
It is possible that Mano want it.